pkg/skaffold/deploy/kubectl.go

```diff
@@ -79,12 +80,18 @@ func (k *KubectlDeployer) Deploy(ctx context.Context, out io.Writer, builds []bu
 		return nil, errors.Wrap(err, "replacing images in manifests")
 	}
 
-	err = kubectl(manifests.reader(), out, k.kubeContext, k.Flags.Global, "apply", k.Flags.Apply, "-f", "-")
+	// Only redeploy modified or new manifests
+	// TODO(dgageot): should we delete a manifest that was deployed and is not anymore?
```

Comment:
My opinion here would be no, because 
1) I like the experience of having a deployment recreated by skaffold if I delete it manually (read: accidentally) through kubectl/cloud console, and
2) this would mean you could get skaffold into a state where it's not consistent with the configuration in the skaffold.yaml, which is a weird UX

Reply:
I understand. This PR will not anymore recreate deployments that you accidentally deleted. In exchange, dev cycles are much faster. wdyt?

Reply:
It's a tradeoff, but I guess if you do something silly like delete a deployment you could always just restart skaffold and have those deployments recreated. I could be convinced that speed is more important here.